test/sha512_test.dart

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:test/test.dart';
+import 'package:crypto/crypto.dart';
+
+void main() {
```

Comment:
perfect! I will add this later tonight and this was exactly the input I was looking for since contrary to popular belief, "this is a test" is not a good test :)

Reply:
Added monte tests for all sha-[224|256|384|512] hashs